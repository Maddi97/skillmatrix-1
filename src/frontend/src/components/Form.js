import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import {
  SkillNameInput, DateInput, TextArea, LevelPicker,
} from './InputFields';

const styles = theme => ({
  root: {
    padding: theme.spacing.unit * 2,
    marginTop: theme.spacing.unit,
    textAlign: 'left',
  },
});

function Label(props) {
  const { classes, value } = props;

  return (
    <div>
      <Paper className={classes.root} elevation={1}>
        <Typography variant="h5" component="h3">
          {value}
        </Typography>
        <Typography component="p">{value}</Typography>
      </Paper>
    </div>
  );
}

Label.propTypes = {
  classes: PropTypes.instanceOf(styles).isRequired,
};

const LabelField = withStyles(styles)(Label);

export default class Form extends Component {
  renderLabels() {
    const { inputs } = this.props;
    const labels = Object.keys(inputs).map((id, i) => (
      <LabelField key={i} name={inputs[id].name} value={inputs[id].value} />
    ));
    console.log(Object.keys(inputs));
    return labels;
  }

  render() {
    const {
      inputs, page, onChange, onReset, onSubmit,
    } = this.props;
    if (page === 'form') {
      return (
        <div className="form">
          <h1>Neuen Skill erstellen</h1>
          <SkillNameInput
            data={inputs.textfield}
            value={inputs.textfield.value}
            onChange={(id, value) => onChange(id, value)}
          />
          <LevelPicker data={inputs.levelfield} onChange={(id, value) => onChange(id, value)} />
          <DateInput
            data={inputs.datefield}
            value={inputs.datefield.value}
            onChange={(id, value) => onChange(id, value)}
          />
          <TextArea
            data={inputs.textarea}
            value={inputs.textarea.value}
            onChange={(id, value) => onChange(id, value)}
          />

          <div className="button-container">
            <Button variant="contained" color="primary" onClick={() => onReset()}>
              Reset Form
            </Button>
            <Button
              variant="contained"
              color="primary"
              name="submit"
              onClick={() => onSubmit('skill')}
            >
              Show skill
            </Button>
          </div>
        </div>
      );
    }
    return (
      <div className="form">
        {this.renderLabels()}
        <div className="button-container">
          <Button
            variant="contained"
            color="primary"
            name="submit"
            onClick={() => onSubmit('form')}
          >
            Edit skill
          </Button>
        </div>
      </div>
    );
  }
}
