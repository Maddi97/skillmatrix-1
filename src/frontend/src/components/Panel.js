import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const styles = theme => ({
  root: {
    width: '500px',
    margin: '2em auto',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '33.33%',
    flexShrink: 0,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

function Panel(props) {
  const {
    classes, onChange, id, isExpanded, username, level,
  } = props;

  console.log(props);
  return (
    <ExpansionPanel
      expanded={isExpanded}
      onChange={(expanded) => {
        onChange(id);
        console.log(id);
        console.log(`exp ${expanded}`);
      }}
    >
      <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
        <Typography className={classes.heading}>{username}</Typography>
        <Typography className={classes.secondaryHeading}>{`Level: ${level}`}</Typography>
      </ExpansionPanelSummary>
      <ExpansionPanelDetails>
        <Typography>
          Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget maximus
          est, id dignissim quam.
        </Typography>
      </ExpansionPanelDetails>
    </ExpansionPanel>
  );
}

Panel.propTypes = {
  classes: PropTypes.instanceOf(styles).isRequired,
};

export default withStyles(styles)(Panel);
