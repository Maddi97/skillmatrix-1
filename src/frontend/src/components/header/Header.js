import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import { AccountCircle, PowerSettingsNew, Search } from '@material-ui/icons';

// import redux parts
import store from '../../Store';
import {
  switchPage,
  setError,
  resetState,
  changeProfileOwner,
  resetSearch,
} from 'actions';

// Rest
import RestPoints from '../../rest/Init';
import RestCom from '../../rest/Rest';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

class ButtonAppBar extends Component {
  static switchToPage(page) {
    store.dispatch(switchPage(page));
  }

  switchToProfilePage = () => {
    store.dispatch(changeProfileOwner(0));
    this.constructor.switchToPage('profile');
  }

  switchToSearchPage = () => {
    store.dispatch(resetSearch);
    this.constructor.switchToPage('search');
  }

  async handleLogout() {
    const { state } = this.props;
    const user = {
      username: state.user,
    };
    const Rest = new RestCom(RestPoints.logout, JSON.stringify(user));
    try {
      await Rest.post();
      // logout in frontend
      // reset state
      store.dispatch(resetState);
      // go back to login
      store.dispatch(switchPage('login'));
    } catch (e) {
      // display error Message to user<
      console.log(e);
      store.dispatch(setError(e.message));
    }
  }

  render() {
    const { classes, user } = this.props;

    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              className={classes.menuButton}
              onClick={this.switchToProfilePage}
              color="inherit"
              aria-label="Menu"
            >
              <AccountCircle />
            </IconButton>
            <Typography variant="h6" color="inherit" className={classes.grow}>
              {user.name}
            </Typography>
            <IconButton
              className={classes.menuButton}
              onClick={this.switchToSearchPage}
              color="inherit"
              aria-label="Menu"
            >
              <Search />
            </IconButton>
            <IconButton
              className={classes.menuButton}
              onClick={() => this.handleLogout()}
              color="inherit"
              aria-label="Menu"
            >
              <PowerSettingsNew />
            </IconButton>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

export default withStyles(styles)(ButtonAppBar);
