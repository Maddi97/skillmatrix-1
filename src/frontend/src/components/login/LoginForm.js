// react
import React, { Component } from 'react';
import ReactDOM from 'react-dom';

// material-ui
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import LockIcon from '@material-ui/icons/LockOutlined';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import withStyles from '@material-ui/core/styles/withStyles';
import CircularProgress from '@material-ui/core/CircularProgress';

// import redux parts
import store from 'Store';
import {
  switchPage, setLoginError, setUser, setOwnProfile,
} from 'actions';

import { updateAllSkills } from 'rest/handleCommonRequests';

// Rest
import RestPoints from 'rest/Init';
import RestCom from 'rest/Rest';

const styles = theme => ({
  main: {
    width: 'auto',
    display: 'block', // Fix IE 11 issue.
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing.unit * 10,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px`,
  },
  avatar: {
    margin: theme.spacing.unit,
    backgroundColor: theme.palette.secondary.main,
  },
  spinner: {
    margin: theme.spacing.unit,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing.unit,
  },
  submit: {
    marginTop: theme.spacing.unit * 3,
  },
});

class SignIn extends Component {
  static async handleLogin(event) {
    // avoid reloading on submit
    event.preventDefault();
    // avoid event is nullified => https://reactjs.org/docs/events.html#event-pooling
    event.persist();

    const { target } = event;

    // ensure immutability
    const usernameInput = { ...target.username };
    const passwordInput = { ...target.password };

    const loginCredentials = {
      username: usernameInput.value,
      password: passwordInput.value,
    };

    const Rest = new RestCom(RestPoints.login, JSON.stringify(loginCredentials));

    try {
      const { user } = await Rest.post();
      const { username, name } = user;
      store.dispatch(setUser({ username, name }));
      store.dispatch(setOwnProfile(user));
      store.dispatch(switchPage('search'));
    } catch (e) {
      // clear password input
      target.password.value = '';
      store.dispatch(setLoginError(e.message));
    }

    await updateAllSkills();
  }

  componentDidMount() {
    // Get the components DOM node
    const elem = ReactDOM.findDOMNode(this);
    // Set the opacity of the element to 0
    elem.style.opacity = 0;
    window.requestAnimationFrame(() => {
      // Now set a transition on the opacity
      elem.style.transition = 'opacity 250ms';
      // and set the opacity to 1
      elem.style.opacity = 1;
    });
  }

  render() {
    const { classes, errorMsg, loading } = this.props;

    return (
      <main className={classes.main}>
        <CssBaseline />
        <Paper className={classes.paper}>
          {loading ? (
            <CircularProgress className={classes.spinner} />
          ) : (
            <Avatar className={classes.avatar}>
              <LockIcon />
            </Avatar>
          )}
          <Typography component="h1" variant="h4">
            Skill Matrix
          </Typography>
          <p className="error">{errorMsg}</p>
          <form onSubmit={this.constructor.handleLogin} className={classes.form}>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="username">Username</InputLabel>
              <Input id="username" name="username" autoFocus />
            </FormControl>
            <FormControl margin="normal" required fullWidth>
              <InputLabel htmlFor="password">Password</InputLabel>
              <Input name="password" type="password" id="password" defaultValue="" />
            </FormControl>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Sign in
            </Button>
          </form>
          <Button
            type="button"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={() => store.dispatch(switchPage('search'))}
          >
            skip login (dev only)
          </Button>
        </Paper>
      </main>
    );
  }
}

export default withStyles(styles)(SignIn);
