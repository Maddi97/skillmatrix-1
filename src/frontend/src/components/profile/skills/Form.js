import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import {
  SkillNameInput, DateInput, TextArea, LevelPicker,
} from 'components/common/InputFields';
import LabelField from './Label';


export default class Form extends Component {
  renderLabels() {
    const { inputs } = this.props;
    const labels = Object.keys(inputs).map((id, i) => (
      <LabelField key={i} name={inputs[id].name} value={inputs[id].value} />
    ));
    console.log(Object.keys(inputs));
    return labels;
  }

  render() {
    const {
      inputs, page, onChange, onReset, onSubmit,
    } = this.props;
    if (page === 'form') {
      return (
        <div className="form">
          <h1>Neuen Skill erstellen</h1>
          <SkillNameInput
            data={inputs.textfield}
            value={inputs.textfield.value}
            onChange={(id, value) => onChange(id, value)}
          />
          <LevelPicker data={inputs.levelfield} onChange={(id, value) => onChange(id, value)} />
          <DateInput
            data={inputs.datefield}
            value={inputs.datefield.value}
            onChange={(id, value) => onChange(id, value)}
          />
          <TextArea
            data={inputs.textarea}
            value={inputs.textarea.value}
            onChange={(id, value) => onChange(id, value)}
          />

          <div className="button-container">
            <Button variant="contained" color="primary" onClick={() => onReset()}>
              Reset Form
            </Button>
            <Button
              variant="contained"
              color="primary"
              name="submit"
              onClick={() => onSubmit('skill')}
            >
              Show skill
            </Button>
          </div>
        </div>
      );
    }
    return (
      <div className="form">
        {this.renderLabels()}
        <div className="button-container">
          <Button
            variant="contained"
            color="primary"
            name="submit"
            onClick={() => onSubmit('form')}
          >
            Edit skill
          </Button>
        </div>
      </div>
    );
  }
}
