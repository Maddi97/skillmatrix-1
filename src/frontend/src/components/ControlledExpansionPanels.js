import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Panel from './Panel';

const styles = theme => ({
  root: {
    width: '500px',
    margin: '2em auto',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '33.33%',
    flexShrink: 0,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

function ControlledExpansionPanels(props) {
  const { classes, results } = props;

  const panels = Object.keys(results).map((username, i) => (
    <Panel
      key={i}
      id={i}
      username={username}
      level={results[username]}
      onChange={() => console.log('test')}
    />
  ));

  return <div className={classes.root}>{panels}</div>;
}

ControlledExpansionPanels.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ControlledExpansionPanels);
