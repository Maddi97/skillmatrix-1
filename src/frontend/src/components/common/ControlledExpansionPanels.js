import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import Panel from 'components/search/Panel';

const styles = theme => ({
  root: {
    width: '500px',
    margin: '2em auto',
  },
  heading: {
    fontSize: theme.typography.pxToRem(15),
    flexBasis: '33.33%',
    flexShrink: 0,
  },
  secondaryHeading: {
    fontSize: theme.typography.pxToRem(15),
    color: theme.palette.text.secondary,
  },
});

function ControlledExpansionPanels(props) {
  const { classes, results, heading } = props;

  const panels = results.map((user, i) => (
    <Panel key={i} id={i} username={user.username} skills={user.skills} />
  ));

  return (
    <div className={classes.root}>
      <h2>{heading}</h2>
      {panels}
    </div>
  );
}

ControlledExpansionPanels.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ControlledExpansionPanels);
