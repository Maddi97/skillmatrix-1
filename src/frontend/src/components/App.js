// import react
import React, { Component } from "react";

// import redux
import store from "../Store";
import {
  updateInput,
  switchPage,
  setInputError,
  setError,
  resetForm,
  setUsername,
  resetState,
  setSearchResults,
  showSearchResults,
  hideSearchResults
} from "../actions";

// import page parts
import Header from "./Header";
import Form from "./Form";
import Search from "./Search";
import LoginForm from "./LoginForm";
import ErrorPaper from "./ErrorPaper";
import ControlledExpansionPanels from "./ControlledExpansionPanels";

// Rest
import RestPoints from "../rest/Init";
import RestCom from "../rest/Rest";

class App extends Component {
  static async handleLogin(username, password) {
    const loginCredentials = {
      username,
      password
    };
    const Rest = new RestCom(
      RestPoints.login,
      JSON.stringify(loginCredentials)
    );

    try {
      const { data } = await Rest.post();
      const { user } = data;
      store.dispatch(setUsername(user.username));
      store.dispatch(switchPage("search"));
    } catch (e) {
      store.dispatch(setError(e.message));
      store.dispatch(switchPage("login"));
    }
  }

  // user wants to reset all input fields
  static handleResetForm() {
    store.dispatch(resetForm);
  }

  constructor(props) {
    super(props);
    // This binding is necessary to make `this` work in the callback
    this.handleLogout = this.handleLogout.bind(this);
  }

  // user inputs something into an input field
  handleChange(id, value) {
    const { state } = this.props;

    console.log(`${id}  ${value}`);

    if (state.formState[id].error) {
      store.dispatch(setInputError(id, false));
    }
    store.dispatch(updateInput(id, value));
  }

  // user clicks button to show form/labels
  handleSubmit(newPage) {
    const { state } = this.props;
    const { page } = state;

    const inputs = state.formState;
    let submit = true;
    if (page === "form") {
      const keys = Object.keys(inputs);
      keys.foreach(key => {
        const input = inputs[key];
        if (input.value === "") {
          store.dispatch(setInputError(key, true));
          submit = false;
        } else {
          store.dispatch(setInputError(key, false));
        }
      });
    }
    if (submit) {
      store.dispatch(switchPage(newPage));
    }
  }

  // get results for query when user clicks on search button and store them into state
  async handleSearch() {
    const { state } = this.props;
    const { user } = state;
    const { value } = state.formState.searchfield;
    const search = {
      username: user,
      query: value
    };
    const Rest = new RestCom(RestPoints.search, JSON.stringify(search));
    try {
      const { data } = await Rest.post();
      // store results into state
      store.dispatch(setSearchResults(data));
      // show results to user
      store.dispatch(showSearchResults);
    } catch (e) {
      store.dispatch(setError(e.message));
      console.log(e);
    }
  }

  async handleLogout() {
    const { state } = this.props;
    const user = {
      user: state.user
    };
    const Rest = new RestCom(RestPoints.logout, JSON.stringify(user));
    try {
      await Rest.post();
      // logout in frontend
      // reset state
      store.dispatch(resetState);
      // go back to login
      store.dispatch(switchPage("login"));
    } catch (e) {
      // display error Message to user<
      console.log(e);
      store.dispatch(setError(e.message));
    }
  }

  render() {
    const { state } = this.props;
    const { page, error, user, formState, searchResults } = state;
    let main = [];

    if (page === "login") {
      return (
        <LoginForm
          errorMsg={error.message}
          login={(username, password) => App.handleLogin(username, password)}
        />
      );
    }
    const { results } = searchResults;
    switch (page) {
      case "form":
        main = (
          <Form
            inputs={formState}
            page={page}
            name="test"
            onChange={(id, value) => this.handleChange(id, value)}
            onSubmit={newPage => this.handleSubmit(newPage)}
            onReset={() => this.handleResetForm()}
          />
        );
        break;
      case "search":
        main.push(
          <Search
            searchField={formState.searchfield}
            onChange={(id, value) => this.handleChange(id, value)}
            onSearch={() => this.handleSearch()}
            key="search"
          />
        );
        if (searchResults.showResults) {
          main.push(
            Object.keys(results).map((category, i) => (
              <ControlledExpansionPanels results={results[category]} key={i} />
            ))
          );
        }
        break;
      default:
        return "Error";
    }

    let errorPaper = "";
    if (error.hasError) {
      errorPaper = <ErrorPaper errorMsg={error.message} />;
    }
    return (
      <div>
        <Header username={user} logout={this.handleLogout} />
        <main>
          {main}
          {errorPaper}
        </main>
      </div>
    );
  }
}

export default App;
