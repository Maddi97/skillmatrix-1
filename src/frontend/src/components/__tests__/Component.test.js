import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../LoginForm';
import Form from '../Form';
import Header from '../Header';
import App from '../App';
import Panel from '../Panel'
import ControlledExpansionPanels from '../ControlledExpansionPanels'
import {
  SkillNameInput, DateInput, TextArea, LevelPicker, SearchInput
} from '../InputFields';

const username = 'Valdemar Forsberg';
const password = 'password';

describe('test Login Components', () => {
  const wrapper = shallow(<LoginForm errorMsg="" login={(username, password) => App.handleLogin(username, password)} />);
  it('render Login component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test Form component', () => {
  const wrapper = shallow(<Form inputs="" page="login" name="test" onChange="" onSubmit="" onClick="" />);
  it('contains important components as children', () => {
    expect(wrapper.contains([<SkillNameInput />, <DateInput />, <TextArea />, <LevelPicker />]));
  });
  it('renders single component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test Header component', () => {
  const wrapper = shallow(<Header username="Valdemar" logout="handleLogout" />);
  it('render Header component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test components from InputFields', () => {
  it('render SkillNameInput from InputFields component', () => {
    const wrapper = shallow(<SkillNameInput data="skill" value="python" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render DateInput from InputFields', () => {
    const wrapper = shallow(<DateInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render LevelPicker from InputFields', () => {
    const wrapper = shallow(<LevelPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render TextArea from InputFields', () => {
    const wrapper = shallow(<TextArea />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render SearchInput from InputFields', () => {
    const wrapper = shallow(<SearchInput />)
    expect(wrapper).toMatchSnapshot();
  });
});


describe('test Panels right', () => {
  const wrapper = shallow(<Panel
    key={1}
    id={1}
    username={username} 
    level={2}
    onChange={() => console.log('test')}
/>);
  it('render Panel component', () => {
    const wrapper = shallow(<Panel
      key={1}
      id={1}
      username={username} 
      level={2}
      onChange={() => console.log('test')}
/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('render ControlledExpansionPanels component', () => {
    const wrapper = shallow(<ControlledExpansionPanels />);
    expect(wrapper).toMatchSnapshot();
  });
});