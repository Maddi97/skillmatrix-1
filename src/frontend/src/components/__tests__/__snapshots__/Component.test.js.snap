// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ExpensionPanels right render ControlledExpansionPanels component 1`] = `
<ControlledExpansionPanels
  classes={
    Object {
      "heading": "ControlledExpansionPanels-heading-25",
      "root": "ControlledExpansionPanels-root-24",
      "secondaryHeading": "ControlledExpansionPanels-secondaryHeading-26",
    }
  }
/>
`;

exports[`test Header component render Header component 1`] = `
<ButtonAppBar
  classes={
    Object {
      "grow": "ButtonAppBar-grow-7",
      "menuButton": "ButtonAppBar-menuButton-8",
      "root": "ButtonAppBar-root-6",
    }
  }
  logout="handleLogout"
  username="Valdemar"
/>
`;

exports[`test Login Components render Login component 1`] = `
<SignIn
  classes={
    Object {
      "avatar": "SignIn-avatar-3",
      "form": "SignIn-form-4",
      "main": "SignIn-main-1",
      "paper": "SignIn-paper-2",
      "submit": "SignIn-submit-5",
    }
  }
  errorMsg=""
  login={[Function]}
/>
`;

exports[`test components from InputFields render DateInput from InputFields 1`] = `
<Component
  classes={
    Object {
      "container": "Component-container-12",
      "error": "Component-error-14",
      "textField": "Component-textField-13",
    }
  }
/>
`;

exports[`test components from InputFields render LevelPicker from InputFields 1`] = `
<Component
  classes={
    Object {
      "container": "Component-container-15",
      "error": "Component-error-17",
      "textField": "Component-textField-16",
    }
  }
/>
`;

exports[`test components from InputFields render SearchInput from InputFields 1`] = `
<Component
  classes={
    Object {
      "container": "Component-container-21",
      "error": "Component-error-23",
      "textField": "Component-textField-22",
    }
  }
/>
`;

exports[`test components from InputFields render SkillNameInput from InputFields component 1`] = `
<Component
  classes={
    Object {
      "container": "Component-container-9",
      "error": "Component-error-11",
      "textField": "Component-textField-10",
    }
  }
  data="skill"
  value="python"
/>
`;

exports[`test components from InputFields render TextArea from InputFields 1`] = `
<Component
  classes={
    Object {
      "container": "Component-container-18",
      "error": "Component-error-20",
      "textField": "Component-textField-19",
    }
  }
/>
`;
